### PR TITLE
add support to ignore volumeless pods by Resiliency

### PR DESCRIPTION
**Pull request by alikdell**
# Description
Ignore volumeless pods with Resiliency label.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| #https://github.com/dell/csm/issues/493 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Test A
- [x] Test B

PASS
coverage: 92.3% of statements
status 0
ok      podmon/internal/monitor 8.636s  coverage: 92.3% of statements

### CODE REVIEW

File: .github/containerscan/allowedlist.yaml [hunk 1]
```diff
@@ -11,10 +11,12 @@ general:
     - CVE-2021-44568
     - CVE-2022-0778
     - CVE-2022-2526
+    - CVE-2022-27664
     # Disputed CVEs
     - CVE-2019-1010022
```
Comment (hoppea2) [inline .github/containerscan/allowedlist.yaml:16]: Should we take this opportunity to review this allowedList for CVEs?

File: internal/monitor/controller.go [hunk 2]
```diff
@@ -366,12 +384,12 @@ func (cm *PodMonitorType) callControllerUnpublishVolume(node *v1.Node, volumeID
...
-// podToArrayIDs returns the array IDs used by the pod)
-// TODO: multi-array
-func (cm *PodMonitorType) podToArrayIDs(pod *v1.Pod) ([]string, error) {
-	arrayIDs := make([]string, 1)
-	arrayIDs[0] = "default"
-	return arrayIDs, nil
+// podToArrayIDs returns the array IDs used by the pod, along with pvCount, and error
+func (cm *PodMonitorType) podToArrayIDs(ctx context.Context, pod *v1.Pod) ([]string, int, error) {
+	arrayIDs := make([]string, 0)
+	pvlist, _ := K8sAPI.GetPersistentVolumesInPod(ctx, pod)
```
Comment (atye) [inline internal/monitor/controller.go:390]: Does this return an error that is being discarded? Should it not be checked?

File: internal/monitor/monitor_steps_test.go [hunk 3]
```diff
@@ -143,6 +143,10 @@ func (f *feature) aPodForNodeWithVolumesCondition(node string, nvolumes int, con
 }
 
 func (f *feature) aPodForNodeWithVolumesConditionAffinity(node string, nvolumes int, condition, affinity string) error {
+	// To test IgnoreVolumelessPods vnvolumes is supplied 0 to induce it volumeless pod
```
Comment (hoppea2) [inline internal/monitor/monitor_steps_test.go:146]: typo? vnvolumes vs nvolumes

File: internal/monitor/monitor_steps_test.go [hunk 4]
```diff
@@ -158,6 +162,10 @@ func (f *feature) aPodForNodeWithVolumesConditionAffinity(node string, nvolumes
 
 func (f *feature) iHaveAPodsForNodeWithVolumesDevicesCondition(nPods int, nodeName string, nvolumes, ndevices int, condition string) error {
 	var err error
+	// To test IgnoreVolumelessPods vnvolumes is supplied 0 to induce it volumeless pod
```
Comment (hoppea2) [inline internal/monitor/monitor_steps_test.go:165]: typo? vnvolumes vs nvolumes

File: internal/monitor/controller.go (hunk 2 above)
Comment (alikdell) [reply internal/monitor/controller.go:390]: Here we don't need, error will result in empty pvlist. We check for error where we need to, here for now not needed. When we add multi-array support, then we will need to check error and accordingly return error.

File: .github/containerscan/allowedlist.yaml (hunk 1 above)
Comment (alikdell) [reply .github/containerscan/allowedlist.yaml:16]: We have ubi image upgrade story, we should check in that story. I will add in acceptance criteria

File: cmd/podmon/main.go [hunk 5]
```diff
@@ -211,6 +213,7 @@ type PodmonArgs struct {
 	driverConfigParamsFile                   *string // Set the location of the driver ConfigMap
 	driverPodLabelKey                        *string // driverPodLabelKey for annotating driver node pods to be watched/processed
 	driverPodLabelValue                      *string // driverPodLabelValue value for annotating driver node pods to be watched/processed
+	ignoreVolumelessPods                     *bool   // Ignore volumeless pods even if they
```
Comment (shaynafinocchiaro) [inline cmd/podmon/main.go:216]: End of comment looks like it got cut off

Comment (alikdell) [reply cmd/podmon/main.go]: Actually according to DRY we do not need that comment as variable is implicit.

**Review by hoppea2 — APPROVED** [review]

**Review by atye — DISMISSED** [review]

**Review by sharmilarama — DISMISSED** [review]

**Review by atye — APPROVED** [review]

**Review by shaynafinocchiaro — APPROVED** [review]

**Pull request merged**